Add activity docs (#2544)
l2beat/l2beat: docs/activity.md

> # How to add new activity module for existing project?

Reviewer: ```suggestion
# How to add activity tracking for existing project?
```

> ### Non starkex type project

Reviewer: remove this heading

> 1. Find a RPC for a project you want to add activity module for. Try to write directly to project to

Reviewer: add hyperlink to starkex there

> 1. Find a RPC for a project you want to add activity module for. Try to write directly to project to

Reviewer: ```suggestion
1. Find RPC endpoint for a project you want to add. Try to write directly to project to
```

>    obtain a private node access. If you can't, you can use a public node.

Reviewer: ```suggestion
   obtain a private node access. If you can't, you can use a public node or RPC provider e.g. QuickNode.
```

> 2. Next you need to test your RPC locally. Use `Cast` for this. If you do not have it yet, just

Reviewer: ```suggestion
2. Next you need to test your RPC locally. Use `cast` for this. If you do not have it yet, just
```

   - `timeout` (optional): timeout property for provider, you do not have to set it up in most cases
   - `startBlock` (optional): if you want to start from specific block, set it here
   - `assessCount` (optional): custom function to assess count of transactions in block

4. (Only if you are using private RPC) - if you are using private RPC do not fill in the `url`
   property in previous step. Instead, go to `config.production.ts` and `config.local.ts` in
   `backend` folder to configure it. Add `activity.projects.<projectId>` property(projectId has to
   match projectId in project config file) and fill it:

```
        <projectId>: {
          type: 'rpc',
          callsPerMinute: env.integer('ACTIVITY_<PROJECT_ID>_CALLS'),
          url: env.string('ACTIVITY_<PROJECT_ID>_URL'),
        },
```

    After that, remember to add these variables to the Heroku config vars before merging them into the master branch

### Starkex type project

1. Go to this page:
   https://app.swaggerhub.com/apis-docs/StarkExBI/StarkExAggregations/0.3#/Queries/post_aggregations_count
   and check if your project is listed in `product` array in endpoint description section. If it is,

>    you can go to the next step. If not, you need to talk to starkex devs to add support of your

Reviewer: ```suggestion
   you can go to the next step. If not, you need to talk to starkex devs to add support for your
```

# How to add activity tracking for existing project? ?

## Introduction

Within L2BEAT💗 we have an [Activity](https://l2beat.com/scaling/activity) tab which tracks activity

> of scaling solutions. 'Activity' in this context refers to the range of transactions that occur on

Reviewer: ```suggestion
of scaling solutions. 'Activity' in this context refers to the amount of transactions that occur on
```

```
        <projectId>: {
          type: 'rpc',
          callsPerMinute: env.integer('ACTIVITY_<PROJECT_ID>_CALLS'),
          url: env.string('ACTIVITY_<PROJECT_ID>_URL'),
        },
```

    After that, remember to add these variables to the Heroku config vars before merging them into the master branch

### Starkex type project

1. Go to this page:
   https://app.swaggerhub.com/apis-docs/StarkExBI/StarkExAggregations/0.3#/Queries/post_aggregations_count
   and check if your project is listed in `product` array in endpoint description section. If it is,
   you can go to the next step. If not, you need to talk to starkex devs to add support for your
   project.

2. Next step is to fill activity config in project config file. Add `config.transactionApi` property
   and fill it:

   - `type` (required): set it as `'starkex'`
   - `product` (required): set it for you project `product` property from swaggerhub
   - `sinceTimestamp` (required): from which timestamp you want to start fetching data
   - `resyncLastDays` (required): set it to 7

## Test locally

1. Set up your backend according to the instructions in the [README](../packages/backend/README.md).

> 2. Add `ACTIVITY_ENABLED=true` to your `.env` file. Also add `STARKEX_API_KEY` which is required.

Reviewer: what about setting envs for project that you want to test?
